cli/src/maintenance.rs

```diff
@@ -439,29 +439,38 @@ impl SolidoState {
 
     /// If there is a validator being deactivated, try to unstake its funds.
     pub fn try_unstake_from_inactive_validator(&self) -> Option<(Instruction, MaintenanceOutput)> {
-        for validator in &self.solido.validators.entries {
-            // We are only interested in unstaking from inactive validators.
-            if validator.entry.active {
+        for (validator, stake_accounts) in self
+            .solido
+            .validators
+            .entries
+            .iter()
+            .zip(self.validator_stake_accounts.iter())
+        {
+            // We are only interested in unstaking from inactive validators that
+            // have stake accounts.
+            if validator.entry.active || stake_accounts.is_empty() {
+                continue;
+            }
+            // Validator's stake account is not fully inactive.
+            if stake_accounts[0].1.balance.inactive != stake_accounts[0].1.balance.total() {
```

Comment:
Small suggestion:  Using the tuple accessors everywhere makes this a little difficult to read (reason about) in my opinion.  An improvement would be to map the stake accounts into a struct with field names.  Or better yet, change:

```Rust
    pub validator_stake_accounts: Vec<Vec<(Pubkey, StakeAccount)>>,
```
to use a struct rather than a ```(Pubkey, StakeAccount)``` but that will have a little more impact.

Reply:
Can also give it a name temporarily:

```rust
let (stake_account_address, stake_account_balance) = &stake_accounts[0];